### DISCUSSION TITLE
Use git-flow?

### DISCUSSION THREAD

**Discussion by Marshall-Hallenbeck** [Pennyw0rth/NetExec#23] [Ideas]
Should we start using git-flow as a branching and release method?

https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow

**Comment by bongobongoland** [comment]
I'm not familiar with git-flow so I find this concerning
_These long-lived feature branches require more collaboration to merge and have a higher risk of deviating from the trunk branch. They can also introduce conflicting updates._

**Comment by NeffIsBack** [comment]
Actually i think we are pretty close to git-flow already, aren't we? Feature branches are pretty much what we are working with but i also thought about the separation of main and dev branch. Maybe this would be worth a try, so the main branch is always a stable version and the dev branch is what we are working with.

**Reply by Marshall-Hallenbeck** [reply]
Yeah, I agree, we should _at least_ start using a dev branch. We pretty much use the feature branches mechanic, but don't use the naming convention really, which I'm fine with.

**Comment by Marshall-Hallenbeck** [comment]
@NeffIsBack I think after v1 we can start using git-flow, at least trying to name our branches correctly and using the develop branch. That way we can more easily tag code, etc.

**Reply by Marshall-Hallenbeck** [reply]
When we do this we can switch the default branch to `develop` as well, so we are merging stuff into that, then on releases (major, minor, and hotfixes) we merge into `main` and tag it.
